bindings/java/CMakeLists.txt

```diff
@@ -256,29 +254,118 @@ if(NOT OPEN_FOR_IDE)
-    file(DOWNLOAD "https://repo1.maven.org/maven2/org/hamcrest/hamcrest-all/1.3/hamcrest-all-1.3.jar"
-      ${CMAKE_BINARY_DIR}/packages/hamcrest-all-1.3.jar
-      EXPECTED_HASH SHA256=4877670629ab96f34f5f90ab283125fcd9acb7e683e66319a68be6eb2cca60de)
-    add_jar(fdb-junit SOURCES ${JAVA_JUNIT_TESTS} INCLUDE_JARS fdb-java ${CMAKE_BINARY_DIR}/packages/junit-4.13.jar)
+    # Sets up the JUnit testing structure to run through ctest
+    #
+    # To add a new junit test, add the class to the JAVA_JUNIT_TESTS variable in `src/tests.cmake`. Note that if you run a Suite, 
+    # ctest will NOT display underlying details of the suite itself, so it's best to avoid junit suites in general. Also,
+    # if you need a different runner other than JUnitCore, you'll have to modify this so be aware.
+    #
+    # To run tests (once built), run:
+    #
+    # ctest .
+    #
+    # from the ${BUILD_DIR}/bindings/java subdirectory. 
+    # 
+    # Note: if you are running from ${BUILD_DIR}, additional tests of the native logic will be run. To avoid these, use
+    #
+    # ctest . -R junit 
+    #
+    # ctest has lots of flexible command options, so be sure to refer to its documentation if you want to do something specific(documentation
+    # can be found at https://cmake.org/cmake/help/v3.19/manual/ctest.1.html)
+
+    add_jar(fdb-junit SOURCES ${JAVA_JUNIT_TESTS} ${JUNIT_RESOURCES} INCLUDE_JARS fdb-java 
+            ${CMAKE_BINARY_DIR}/packages/junit-jupiter-api-5.7.1.jar 
+            ${CMAKE_BINARY_DIR}/packages/junit-jupiter-engine-5.7.1.jar
+            ${CMAKE_BINARY_DIR}/packages/junit-jupiter-params-5.7.1.jar
+            ${CMAKE_BINARY_DIR}/packages/opentest4j-1.2.0.jar
+            ${CMAKE_BINARY_DIR}/packages/apiguardian-api-1.1.1.jar
+    )
     get_property(junit_jar_path TARGET fdb-junit PROPERTY JAR_FILE)
-    add_test(NAME junit
+    
+    add_test(NAME JUNIT
       COMMAND ${Java_JAVA_EXECUTABLE}
-              -cp "${target_jar}:${junit_jar_path}:${CMAKE_BINARY_DIR}/packages/junit-4.13.jar:${CMAKE_BINARY_DIR}/packages/hamcrest-all-1.3.jar"
-              -Djava.library.path=${CMAKE_BINARY_DIR}/lib
-              org.junit.runner.JUnitCore "com.apple.foundationdb.tuple.AllTests")
+            -classpath "${target_jar}:${junit_jar_path}:${JUNIT_CLASSPATH}"
+            -Djava.library.path=${CMAKE_BINARY_DIR}/lib
+            org.junit.platform.console.ConsoleLauncher "--details=summary" "-class-path=${junit_jar_path}" "--scan-classpath" "--disable-banner"
+    )
+
+  endif()
+
+  if(RUN_JAVA_INTEGRATION_TESTS)
+    # Set up the integration tests. These tests generally require a running database server to function properly
+    #
+    # Integration tests are run through ctest the same way as unit tests, but their label is prefixed with the entry 'integration-'.
+    # Note that most java integration tests will be skipped if they can't quickly connect to a running FDB instance(depending on how the test is written, anyway). 
```

Comment:
This needs a documentation fix. Part of the move to JUnit 5 involved changing this behavior--if you tell the integration tests to expect a server to be up (i.e. `run.integration.tests=true`), then the tests will fail if they cannot connect to FDB, rather than being skipped. For developer convenience, though, `run.integration.tests=false` will cause the runner to skip them.

For CI, just setting `run.integration.tests=true` (or leaving it off entirely, the default is true) will cause these tests to fail if the FDB server isn't running.

Will adjust the documentation here.